File: aat-lib/src/main/java/ch/bailu/aat_lib/description/AverageSpeedDescription.java
package ch.bailu.aat_lib.description;

import ch.bailu.aat_lib.gpx.GpxInformation;
import ch.bailu.aat_lib.preferences.StorageInterface;
import ch.bailu.aat_lib.resources.Res;

public class AverageSpeedDescription extends SpeedDescription {

    public AverageSpeedDescription(StorageInterface storage) {
        super(storage);
    }

    @Override
    public String getLabel() {
        return Res.str().average();
    }

    @Override
    public void onContentUpdated(int iid, GpxInformation info) {
        setCache(info.getSpeed());
    }
}

File: aat-lib/src/main/java/ch/bailu/aat_lib/description/AverageSpeedDescription.kt
package ch.bailu.aat_lib.description

import ch.bailu.aat_lib.gpx.GpxInformation
import ch.bailu.aat_lib.preferences.StorageInterface
import ch.bailu.aat_lib.resources.Res

open class AverageSpeedDescription(storage: StorageInterface) : SpeedDescription(storage) {
    override fun getLabel(): String {
        return Res.str().average()
    }

    override fun onContentUpdated(iid: Int, info: GpxInformation) {
        setCache(info.getSpeed())
    }
}
